geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/SocketMessageWriter.java

```diff
@@ -45,46 +47,41 @@ public void writeHandshakeMessage(DataOutputStream dos, byte type, String p_msg,
       msg = "";
     }
     dos.writeUTF(msg);
-    if (clientVersion != null && clientVersion.isNotOlderThan(KnownVersion.GFE_61)) {
-      // get all the instantiators.
-      Instantiator[] instantiators = InternalInstantiator.getInstantiators();
-      HashMap instantiatorMap = new HashMap();
-      if (instantiators != null && instantiators.length > 0) {
-        for (Instantiator instantiator : instantiators) {
-          ArrayList instantiatorAttributes = new ArrayList();
-          instantiatorAttributes.add(instantiator.getClass().toString().substring(6));
-          instantiatorAttributes.add(instantiator.getInstantiatedClass().toString().substring(6));
-          instantiatorMap.put(instantiator.getId(), instantiatorAttributes);
-        }
+
```

Comment:
It appears to me (from looking at `ClientCacheNotifier`) that `clientVersion` is _probably_ never `null`. Are you confident that the null check was superfluous?

It's _almost_ not used anymore. But it is still used.

Reply:
I am confident that if it was `null` we would have much bigger problems since all current client versions expect this content to be sent. 

We could change the logic for the check at the end against `GEODE_1_5_0` to account for `null`.

Reply:
Even there though, the clients expecting this `int` will get pretty upset if they don't get it.

Reply:
I think it is best that an exception occur here rather than unknown behavior from silently missing `int` in the message.

Reply:
I have manually verified that `clientVersion` comes from `ClientRegistrationMetadata` which acquires it in `getAndValidateClientVersion()` called from `initialize()` which is called before this method (`writeHandshakeMessage`). When there is a failure to get `clientVersion` this method is not called.

I see a [Geode dev list discussion from 2017](https://markmail.org/search/?q=list%3Aorg.apache.geode.dev+order%3Adate-backward+notnull#query:list%3Aorg.apache.geode.dev%20order%3Adate-backward%20notnull+page:1+mid:daov733nyw5xbhps+state:results) where use of the JetBrains `@NotNull` annotations were discussed with a couple folks agreeing they would be good for Geode. I approve this code as-is, but if you'd like to add some `@NotNull` annos here that would be worth an extra slice of cake 🍰 .